src/test/java/bio/terra/workspace/service/get/GetServiceTest.java

```diff
@@ -0,0 +1,141 @@
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+
+@Tag("unit")
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = Main.class)
+@SpringBootTest
+@AutoConfigureMockMvc
+public class GetServiceTest {
+
+  @Autowired private MockMvc mvc;
+
+  @MockBean private SamService mockSamService;
+
+  // Mock MVC doesn't populate the fields used to build this.
+  @MockBean private AuthenticatedUserRequestFactory mockAuthenticatedUserRequestFactory;
+
+  @Autowired private ObjectMapper objectMapper;
+
+  @Autowired private CreateService createService;
+
+  @Autowired private GetService getService;
+
+  @BeforeEach
+  public void setup() {
+    doNothing().when(mockSamService).createWorkspaceWithDefaults(any(), any());
+    AuthenticatedUserRequest fakeAuthentication = new AuthenticatedUserRequest();
+    fakeAuthentication
+        .token(Optional.of("fake-token"))
+        .email("fake@email.com")
+        .subjectId("fakeID123");
+    when(mockAuthenticatedUserRequestFactory.from(any())).thenReturn(fakeAuthentication);
+  }
+
+  @Test
+  public void testGetMissingWorkspace() throws Exception {
+    MvcResult callResult =
+        mvc.perform(get("/api/v1/workspaces/" + "fake-id")).andExpect(status().is(404)).andReturn();
+
+    ErrorReport error =
+        objectMapper.readValue(callResult.getResponse().getContentAsString(), ErrorReport.class);
+    assertThat(error.getStatusCode(), equalTo(HttpStatus.NOT_FOUND.value()));
+  }
+
+  @Test
+  public void testGetExistingWorkspace() throws Exception {
+    CreateWorkspaceRequestBody body = new CreateWorkspaceRequestBody();
+    UUID workspaceId = UUID.randomUUID();
+    body.setId(workspaceId);
+    body.setAuthToken("fake-user-auth-token");
+    body.setSpendProfile(JsonNullable.undefined());
+    body.setPolicies(JsonNullable.undefined());
+    JobControl jobControl = new JobControl();
+    String jobId = UUID.randomUUID().toString();
+    jobControl.setJobid(jobId);
+    body.setJobControl(jobControl);
+
+    CreatedWorkspace workspace = runCreateWorkspaceCall(body, jobId);
+
+    assertThat(workspace.getId(), not(blankOrNullString()));
+
+    MvcResult callResult =
+        mvc.perform(get("/api/v1/workspaces/" + workspace.getId()))
+            .andExpect(status().is(200))
+            .andReturn();
+
+    WorkspaceDescription desc =
+        objectMapper.readValue(
+            callResult.getResponse().getContentAsString(), WorkspaceDescription.class);
+
+    assertThat(desc.getId(), equalTo(workspaceId));
+  }
+
+  private CreatedWorkspace runCreateWorkspaceCall(CreateWorkspaceRequestBody request, String jobId)
```

Comment:
I lifted this and the below code from `CreateServiceTest`. I can see this code being broadly useful so may pull it out into a common class a la `TestUtils.java`